// library/DotIndicator.js
/* 
		WARNING 
		Auto generated file. 
		Do not modify its contents.
*/

const extend = require('js-base/core/extend');
const FlexLayout = require('sf-core/ui/flexlayout');



const getCombinedStyle = require("library/styler-builder").getCombinedStyle;

const DotIndicator = extend(FlexLayout)(
	//constructor
	function(_super, props) {
		// initalizes super class for this component scope
		_super(this, props);

		const dot0Style = getCombinedStyle(".flexLayout .flexLayout-dotIndicator-item.active", {});  
		var dot0 = new FlexLayout(dot0Style);  
		this.addChild(dot0);
		
		const dot1Style = getCombinedStyle(".flexLayout .flexLayout-dotIndicator-item.inactive", {});  
		var dot1 = new FlexLayout(dot1Style);  
		this.addChild(dot1);
		
		const dot2Style = getCombinedStyle(".flexLayout .flexLayout-dotIndicator-item.inactive", {});  
		var dot2 = new FlexLayout(dot2Style);  
		this.addChild(dot2);
		
		//assign the children to dotIndicator 
		this.children = Object.assign({}, {
			dot0: dot0,
			dot1: dot1,
			dot2: dot2
		});
		
	});

DotIndicator.defaults = getCombinedStyle(".flexLayout .flexLayout-dotIndicator-main", {
	width: 42,
	height: 10,
	positionType: FlexLayout.PositionType.RELATIVE
});

module && (module.exports = DotIndicator);
