// library/Flexlayout354.js
/* 
		WARNING 
		Auto generated file. 
		Do not modify its contents.
*/

const extend = require('js-base/core/extend');
const Label = require('sf-core/ui/label');
const Color = require('sf-core/ui/color');
const TextAlignment = require('sf-core/ui/textalignment');
const Font = require('sf-core/ui/font');
const FlexLayout = require('sf-core/ui/flexlayout');



const getCombinedStyle = require("library/styler-builder").getCombinedStyle;

const Flexlayout354 = extend(FlexLayout)(
	//constructor
	function(_super, props) {
		// initalizes super class for this component scope
		_super(this, props);

		const label344Style = getCombinedStyle(".label", {
			width: 50,
			height: 50,
			backgroundColor: Color.create(0, 255, 255, 255),
			textColor: Color.create(255, 203, 204, 208),
			text: "+",
			textAlignment: TextAlignment.MIDCENTER,
			font: Font.create("Times New Roman", 50, Font.NORMAL)
		});
		var label344 = new Label(label344Style);
		this.addChild(label344);
		
		//assign the children to flexlayout354 
		this.children = Object.assign({}, {
			label344: label344
		});
		
	});

Flexlayout354.defaults = getCombinedStyle(".flexLayout", {
	width: 65,
	height: 65,
	backgroundColor: Color.create(0, 255, 255, 255),
	borderColor: Color.create(255, 203, 204, 208),
	borderRadius: null,
	borderWidth: 0.5,
	alignItems: FlexLayout.AlignItems.CENTER,
	justifyContent: FlexLayout.JustifyContent.CENTER
});

module && (module.exports = Flexlayout354);
