// library/Flexlayout2.js
/* 
		WARNING 
		Auto generated file. 
		Do not modify its contents.
*/

const extend = require('js-base/core/extend');
const FlexLayout = require('sf-core/ui/flexlayout');



const getCombinedStyle = require("library/styler-builder").getCombinedStyle;

const Flexlayout2 = extend(FlexLayout)(
	//constructor
	function(_super, props) {
		// initalizes super class for this component scope
		_super(this, props);

		var flexlayout3 = new FlexLayout(getCombinedStyle(".flexLayout", {
			height: null,
			width: null,
			flexGrow: 1
		})); 
		this.addChild(flexlayout3);
		
		var flexlayout4 = new FlexLayout(getCombinedStyle(".flexLayout", {
			height: 50,
			width: null,
			alignSelf: FlexLayout.AlignSelf.AUTO
		})); 
		this.addChild(flexlayout4);
		
		//assign the children to flexlayout2 
		this.children = Object.assign({}, {
			flexlayout3: flexlayout3,
			flexlayout4: flexlayout4
		});
		
	});

Flexlayout2.defaults = getCombinedStyle(".flexLayout", {
	left: 8,
	top: 392,
	width: 314,
	height: 106
});

module && (module.exports = Flexlayout2);
