// library/HiearchyLevel3Remaining.js
/* 
		WARNING 
		Auto generated file. 
		Do not modify its contents.
*/

const extend = require('js-base/core/extend');
const ImageView = require('sf-core/ui/imageview');
const Color = require('sf-core/ui/color');
const Image = require('sf-core/ui/image');
const ImageFillType = require('sf-core/ui/imagefilltype');
const FlexLayout = require('sf-core/ui/flexlayout');
const Label = require('sf-core/ui/label');
const Font = require('sf-core/ui/font');



const getCombinedStyle = require("library/styler-builder").getCombinedStyle;

const HiearchyLevel3Remaining = extend(FlexLayout)(
	//constructor
	function(_super, props) {
		// initalizes super class for this component scope
		_super(this, props);

		const imageview10Style = getCombinedStyle(".imageView", {
			width: 60,
			height: 60,
			marginLeft: 135,
			borderRadius: 30,
			borderColor: Color.create(255, 151, 151, 151),
			borderWidth: 1,
			image: Image.createFromFile("images://avatar4.png"),
			imageFillType: ImageFillType.ASPECTFIT
		});
		var imageview10 = new ImageView(imageview10Style);
		this.addChild(imageview10);
		
		const flexlayout1092Style = getCombinedStyle(".flexLayout", {
			height: 60,
			width: null,
			paddingLeft: 10,
			flexGrow: 1
		});
		var flexlayout1092 = new FlexLayout(flexlayout1092Style);
		this.addChild(flexlayout1092);
		
		const flexlayout1171Style = getCombinedStyle(".flexLayout", {
			width: 1,
			height: 8,
			top: 0,
			left: 165,
			backgroundColor: Color.create(255, 151, 151, 151),
			positionType: FlexLayout.PositionType.ABSOLUTE
		});
		var flexlayout1171 = new FlexLayout(flexlayout1171Style);
		this.addChild(flexlayout1171);
		
		const flexlayout1185Style = getCombinedStyle(".flexLayout", {
			width: 1,
			height: 8,
			left: 165,
			bottom: 0,
			backgroundColor: Color.create(255, 151, 151, 151),
			positionType: FlexLayout.PositionType.ABSOLUTE
		});
		var flexlayout1185 = new FlexLayout(flexlayout1185Style);
		this.addChild(flexlayout1185);
		
		const label2378Style = getCombinedStyle(".label .label-list-item-value", {
			height: 30,
			width: null,
			text: "Belly Edwards",
			textColor: Color.create(255, 0, 0, 0),
			font: Font.create("Arial", 18, Font.NORMAL)
		});
		var label2378 = new Label(label2378Style);
		if(label2378Style.scrollEnabled === false)
			label2378.ios && (label2378.ios.scrollEnabled = false);
		flexlayout1092.addChild(label2378);
		
		const label2379Style = getCombinedStyle(".label .label-list-item-value", {
			height: 30,
			width: null,
			text: "Sales Responsible"
		});
		var label2379 = new Label(label2379Style);
		if(label2379Style.scrollEnabled === false)
			label2379.ios && (label2379.ios.scrollEnabled = false);
		flexlayout1092.addChild(label2379);
		
		//assign the children to hiearchyLevel3Remaining 
		this.children = Object.assign({}, {
			imageview10: imageview10,
			flexlayout1092: flexlayout1092,
			flexlayout1171: flexlayout1171,
			flexlayout1185: flexlayout1185
		});
		
		//assign the children of flexlayout1092
		flexlayout1092.children = Object.assign({}, {
			label2378: label2378,
			label2379: label2379
		});
		
	});

HiearchyLevel3Remaining.defaults = getCombinedStyle(".flexLayout", {
	width: 360,
	height: 75,
	flexDirection: FlexLayout.FlexDirection.ROW,
	alignItems: FlexLayout.AlignItems.CENTER,
	positionType: FlexLayout.PositionType.RELATIVE
});

module && (module.exports = HiearchyLevel3Remaining);
