// library/HorizontalDivider.js
/* 
		WARNING 
		Auto generated file. 
		Do not modify its contents.
*/

const extend = require('js-base/core/extend');
const FlexLayout = require('sf-core/ui/flexlayout');
const Color = require('sf-core/ui/color');



const getCombinedStyle = require("library/styler-builder").getCombinedStyle;

const HorizontalDivider = extend(FlexLayout)(
	//constructor
	function(_super, props) {
		// initalizes super class for this component scope
		_super(this, props);

		const flexlayout6Style = getCombinedStyle(".flexLayout", {
			backgroundColor: Color.create(255, 151, 151, 151),
			height: null,
			width: null,
			marginLeft: 10,
			marginRight: 10,
			alpha: 0.4,
			flexGrow: 1
		});  
		var flexlayout6 = new FlexLayout(flexlayout6Style);  
		this.addChild(flexlayout6);
		
		//assign the children to horizontalDivider 
		this.children = Object.assign({}, {
			flexlayout6: flexlayout6
		});
		
	});

HorizontalDivider.defaults = getCombinedStyle(".flexLayout", {
	width: 200,
	height: 1,
	backgroundColor: Color.create(255, 255, 255, 255),
	alpha: 1,
	flexGrow: 1,
	positionType: FlexLayout.PositionType.RELATIVE
});

module && (module.exports = HorizontalDivider);
