// library/ItemFile.js
/* 
		WARNING 
		Auto generated file. 
		Do not modify its contents.
*/

const extend = require('js-base/core/extend');
const FlexLayout = require('sf-core/ui/flexlayout');
const Color = require('sf-core/ui/color');
const ImageView = require('sf-core/ui/imageview');
const Image = require('sf-core/ui/image');
const ImageFillType = require('sf-core/ui/imagefilltype');



const getCombinedStyle = require("library/styler-builder").getCombinedStyle;

const ItemFile = extend(FlexLayout)(
	//constructor
	function(_super, props) {
		// initalizes super class for this component scope
		_super(this, props);

		const filePreviewStyle = getCombinedStyle(".imageView", {
			width: null,
			height: null,
			marginTop: 5,
			marginBottom: 5,
			marginLeft: 10,
			marginRight: 10,
			image: Image.createFromFile("images://icon_add_file.png"),
			imageFillType: ImageFillType.ASPECTFIT,
			backgroundColor: Color.create(0, 255, 255, 255),
			flexGrow: 1,
			alignSelf: FlexLayout.AlignSelf.STRETCH
		});  
		var filePreview = new ImageView(filePreviewStyle);  
		this.addChild(filePreview);
		this.filePreview = filePreview;

		//assign the children to itemFile 
		this.children = Object.assign({}, {
			filePreview: filePreview
		});
		
	});

ItemFile.defaults = getCombinedStyle(".flexLayout", {
	width: 65,
	height: 65,
	backgroundColor: Color.create(0, 255, 255, 255),
	borderColor: Color.create(255, 203, 204, 208),
	borderRadius: null,
	borderWidth: 0.5,
	alignItems: FlexLayout.AlignItems.CENTER,
	justifyContent: FlexLayout.JustifyContent.CENTER
});

module && (module.exports = ItemFile);
