// library/ItemExpense.js
/* 
		WARNING 
		Auto generated file. 
		Do not modify its contents.
*/

const extend = require('js-base/core/extend');
const FlexLayout = require('sf-core/ui/flexlayout');
const Color = require('sf-core/ui/color');
const ImageView = require('sf-core/ui/imageview');
const Image = require('sf-core/ui/image');
const ImageFillType = require('sf-core/ui/imagefilltype');
const Label = require('sf-core/ui/label');
const TextAlignment = require('sf-core/ui/textalignment');
const Font = require('sf-core/ui/font');

const HorizontalDivider = require("../../components/HorizontalDivider");

const getCombinedStyle = require("library/styler-builder").getCombinedStyle;

const ItemExpense = extend(FlexLayout)(
	//constructor
	function(_super, props) {
		// initalizes super class for this component scope
		_super(this, props);

		const flexlayout334Style = getCombinedStyle(".flexLayout", {
			width: null,
			height: null,
			paddingLeft: 10,
			paddingRight: 10,
			flexGrow: 1,
			alignSelf: FlexLayout.AlignSelf.STRETCH,
			flexDirection: FlexLayout.FlexDirection.ROW,
			alignItems: FlexLayout.AlignItems.CENTER
		});
		var flexlayout334 = new FlexLayout(flexlayout334Style);
		this.addChild(flexlayout334);
		
		const flexlayout336Style = getCombinedStyle(".flexLayout", {
			left: 0,
			top: 0,
			width: null,
			height: 1,
			backgroundColor: Color.create(255, 255, 255, 255),
			alpha: 1,
			flexGrow: null,
			positionType: FlexLayout.PositionType.RELATIVE,
			alignSelf: FlexLayout.AlignSelf.STRETCH
		});
		var flexlayout336 = new HorizontalDivider(flexlayout336Style, "itemExpense");
		this.addChild(flexlayout336);
		
		const imageview42Style = getCombinedStyle(".imageView", {
			width: 30,
			height: 30,
			image: Image.createFromFile("images://icon_accepted.png"),
			imageFillType: ImageFillType.ASPECTFIT
		});
		var imageview42 = new ImageView(imageview42Style);
		flexlayout334.addChild(imageview42);
		
		const flexlayout32_1Style = getCombinedStyle(".flexLayout", {
			height: null,
			width: null,
			paddingLeft: 20,
			flexGrow: 1,
			alignSelf: FlexLayout.AlignSelf.STRETCH
		});
		var flexlayout32_1 = new FlexLayout(flexlayout32_1Style);
		flexlayout334.addChild(flexlayout32_1);
		
		const flexlayout32Style = getCombinedStyle(".flexLayout", {
			height: null,
			width: null,
			paddingLeft: null,
			flexGrow: 1,
			alignSelf: FlexLayout.AlignSelf.STRETCH
		});
		var flexlayout32 = new FlexLayout(flexlayout32Style);
		flexlayout334.addChild(flexlayout32);
		
		const flexlayout33Style = getCombinedStyle(".flexLayout", {
			width: 25,
			height: 30,
			flexGrow: null,
			justifyContent: FlexLayout.JustifyContent.FLEX_END,
			flexDirection: FlexLayout.FlexDirection.ROW
		});
		var flexlayout33 = new FlexLayout(flexlayout33Style);
		flexlayout334.addChild(flexlayout33);
		
		const label57_2Style = getCombinedStyle(".label .label-list-item-value", {
			width: null,
			height: null,
			text: "$88",
			backgroundColor: Color.create(255, 255, 255, 255),
			textColor: Color.create(255, 0, 0, 0),
			textAlignment: TextAlignment.BOTTOMLEFT,
			flexGrow: 1,
			font: Font.create("Arial", 22, Font.NORMAL)
		});
		var label57_2 = new Label(label57_2Style);
		flexlayout32_1.addChild(label57_2);
		
		const label57Style = getCombinedStyle(".label .label-list-item-value", {
			width: null,
			height: null,
			text: "998230084534",
			textAlignment: TextAlignment.BOTTOMRIGHT,
			textColor: Color.create(255, 0, 0, 0),
			flexGrow: 1,
			font: Font.create("Arial", 14, Font.NORMAL)
		});
		var label57 = new Label(label57Style);
		flexlayout32.addChild(label57);
		
		const imageview3Style = getCombinedStyle(".imageView", {
			width: 15,
			height: 15,
			imageFillType: ImageFillType.ASPECTFIT,
			image: Image.createFromFile("images://icon_attachment.png")
		});
		var imageview3 = new ImageView(imageview3Style);
		flexlayout33.addChild(imageview3);
		
		const label57_1_1Style = getCombinedStyle(".label .label-list-item-value", {
			width: null,
			height: null,
			text: "Travel",
			flexGrow: 1,
			font: Font.create("Arial", 14, Font.NORMAL)
		});
		var label57_1_1 = new Label(label57_1_1Style);
		flexlayout32_1.addChild(label57_1_1);
		
		const label57_1Style = getCombinedStyle(".label .label-list-item-value", {
			width: null,
			height: null,
			text: " 07.05.2017",
			textAlignment: TextAlignment.MIDRIGHT,
			flexGrow: 1,
			font: Font.create("Arial", 14, Font.NORMAL)
		});
		var label57_1 = new Label(label57_1Style);
		flexlayout32.addChild(label57_1);
		
		//assign the children to itemExpense 
		this.children = Object.assign({}, {
			flexlayout334: flexlayout334,
			flexlayout336: flexlayout336
		});
		
		//assign the children of flexlayout334
		flexlayout334.children = Object.assign({}, {
			imageview42: imageview42,
			flexlayout32_1: flexlayout32_1,
			flexlayout32: flexlayout32,
			flexlayout33: flexlayout33
		});
		
		//assign the children of flexlayout32_1
		flexlayout32_1.children = Object.assign({}, {
			label57_2: label57_2,
			label57_1_1: label57_1_1
		});
		
		//assign the children of flexlayout32
		flexlayout32.children = Object.assign({}, {
			label57: label57,
			label57_1: label57_1
		});
		
		//assign the children of flexlayout33
		flexlayout33.children = Object.assign({}, {
			imageview3: imageview3
		});
		
	});

ItemExpense.defaults = getCombinedStyle(".flexLayout", {
	width: 342,
	height: 75,
	flexDirection: FlexLayout.FlexDirection.COLUMN,
	alignItems: FlexLayout.AlignItems.CENTER
});

module && (module.exports = ItemExpense);
