// library/ItemDocument.js
/* 
		WARNING 
		Auto generated file. 
		Do not modify its contents.
*/

const extend = require('js-base/core/extend');
const FlexLayout = require('sf-core/ui/flexlayout');
const Color = require('sf-core/ui/color');
const ImageView = require('sf-core/ui/imageview');
const Image = require('sf-core/ui/image');
const ImageFillType = require('sf-core/ui/imagefilltype');
const Label = require('sf-core/ui/label');
const TextAlignment = require('sf-core/ui/textalignment');
const Font = require('sf-core/ui/font');



const getCombinedStyle = require("library/styler-builder").getCombinedStyle;

const ItemDocument = extend(FlexLayout)(
	//constructor
	function(_super, props) {
		// initalizes super class for this component scope
		_super(this, props);

		const flexlayout276Style = getCombinedStyle(".flexLayout", {
			width: null,
			height: null,
			id: 2,
			flexGrow: 1,
			flexDirection: FlexLayout.FlexDirection.ROW,
			justifyContent: FlexLayout.JustifyContent.CENTER,
			alignItems: FlexLayout.AlignItems.CENTER
		});
		var flexlayout276 = new FlexLayout(flexlayout276Style);
		this.addChild(flexlayout276);
		
		const flexlayout277Style = getCombinedStyle(".flexLayout", {
			width: null,
			height: 1,
			backgroundColor: Color.create(102, 151, 151, 151),
			id: 1
		});
		var flexlayout277 = new FlexLayout(flexlayout277Style);
		this.addChild(flexlayout277);
		
		const itemDocumentImageStyle = getCombinedStyle(".imageView", {
			width: 30,
			height: 30,
			image: Image.createFromFile("images://icon_accepted.png"),
			imageFillType: ImageFillType.ASPECTFIT,
			id: 6
		});
		var itemDocumentImage = new ImageView(itemDocumentImageStyle);
		flexlayout276.addChild(itemDocumentImage);
		this.itemDocumentImage = itemDocumentImage;

		const itemDocumentTitleStyle = getCombinedStyle(".label", {
			text: "Passport",
			width: null,
			height: 60,
			marginLeft: 10,
			id: 5,
			flexGrow: 1
		});
		var itemDocumentTitle = new Label(itemDocumentTitleStyle);
		if(itemDocumentTitleStyle.scrollEnabled === false)
			itemDocumentTitle.ios && (itemDocumentTitle.ios.scrollEnabled = false);
		flexlayout276.addChild(itemDocumentTitle);
		this.itemDocumentTitle = itemDocumentTitle;

		const imageview64Style = getCombinedStyle(".imageView", {
			width: 28,
			height: 14,
			image: Image.createFromFile("images://arrow_right.png"),
			imageFillType: ImageFillType.ASPECTFIT,
			id: 4
		});
		var imageview64 = new ImageView(imageview64Style);
		flexlayout276.addChild(imageview64);
		
		const itemDocumentAcceptanceStyle = getCombinedStyle(".label", {
			backgroundColor: Color.create(199, 241, 241, 241),
			textColor: Color.create(255, 123, 206, 69),
			width: 90,
			height: 20,
			textAlignment: TextAlignment.MIDCENTER,
			marginRight: null,
			marginTop: null,
			right: 0,
			top: 0,
			text: "ACCEPTED",
			id: 3,
			positionType: FlexLayout.PositionType.ABSOLUTE,
			font: Font.create("Arial", 12, Font.NORMAL)
		});
		var itemDocumentAcceptance = new Label(itemDocumentAcceptanceStyle);
		if(itemDocumentAcceptanceStyle.scrollEnabled === false)
			itemDocumentAcceptance.ios && (itemDocumentAcceptance.ios.scrollEnabled = false);
		flexlayout276.addChild(itemDocumentAcceptance);
		this.itemDocumentAcceptance = itemDocumentAcceptance;

		//assign the children to itemDocument 
		this.children = Object.assign({}, {
			flexlayout276: flexlayout276,
			flexlayout277: flexlayout277
		});
		
		//assign the children of flexlayout276
		flexlayout276.children = Object.assign({}, {
			itemDocumentImage: itemDocumentImage,
			itemDocumentTitle: itemDocumentTitle,
			imageview64: imageview64,
			itemDocumentAcceptance: itemDocumentAcceptance
		});
		
	});

ItemDocument.defaults = getCombinedStyle(".flexLayout", {
	width: 360,
	height: 75,
	paddingLeft: 10,
	paddingRight: 10
});

module && (module.exports = ItemDocument);
