// library/ItemSpeech.js
/* 
		WARNING 
		Auto generated file. 
		Do not modify its contents.
*/

const extend = require('js-base/core/extend');
const ImageView = require('sf-core/ui/imageview');
const ImageFillType = require('sf-core/ui/imagefilltype');
const Image = require('sf-core/ui/image');
const FlexLayout = require('sf-core/ui/flexlayout');
const Color = require('sf-core/ui/color');

const getCombinedStyle = require("library/styler-builder").getCombinedStyle;

const ItemSpeech = extend(FlexLayout)(
	//constructor
	function(_super, props) {
		// initalizes super class for this component scope
		_super(this, props);

		const iconStyle = getCombinedStyle(".imageView", {
			width: null,
			height: null,
			imageFillType: ImageFillType.ASPECTFIT,
			image: Image.createFromFile("images://icon_microphone.png"),
			flexGrow: 1
		});
		var icon = new ImageView(iconStyle);
		this.addChild(icon);
		this.icon = icon;

		//assign the children to itemSpeech 
		this.children = Object.assign({}, {
			icon: icon
		});
		
	});

ItemSpeech.defaults = getCombinedStyle(".flexLayout", {
	backgroundColor: Color.create(0, 255, 255, 255),
	width: 14,
	height: 25
});

module && (module.exports = ItemSpeech);
