// library/ItemBirthday.js
/* 
		WARNING 
		Auto generated file. 
		Do not modify its contents.
*/

const extend = require('js-base/core/extend');
const FlexLayout = require('sf-core/ui/flexlayout');
const Color = require('sf-core/ui/color');
const ImageView = require('sf-core/ui/imageview');
const Image = require('sf-core/ui/image');
const ImageFillType = require('sf-core/ui/imagefilltype');
const Label = require('sf-core/ui/label');
const Font = require('sf-core/ui/font');
const TextAlignment = require('sf-core/ui/textalignment');



const getCombinedStyle = require("library/styler-builder").getCombinedStyle;

const ItemBirthday = extend(FlexLayout)(
	//constructor
	function(_super, props) {
		// initalizes super class for this component scope
		_super(this, props);

		const flexlayout489Style = getCombinedStyle(".flexLayout", {
			width: null,
			height: null,
			id: 8,
			paddingLeft: null,
			paddingRight: null,
			flexGrow: 1,
			flexDirection: FlexLayout.FlexDirection.ROW,
			alignItems: FlexLayout.AlignItems.CENTER
		});  
		var flexlayout489 = new FlexLayout(flexlayout489Style);  
		this.addChild(flexlayout489);
		
		const ibDividerStyle = getCombinedStyle(".flexLayout", {
			width: null,
			height: 1,
			marginLeft: null,
			marginRight: null,
			backgroundColor: Color.create(102, 151, 151, 151),
			id: 9
		});  
		var ibDivider = new FlexLayout(ibDividerStyle);  
		this.addChild(ibDivider);
		
		const ibUserImageStyle = getCombinedStyle(".imageView", {
			width: 60,
			height: 60,
			marginLeft: 5,
			image: Image.createFromFile("images://avatar1.png"),
			imageFillType: ImageFillType.ASPECTFIT,
			id: 7
		});  
		var ibUserImage = new ImageView(ibUserImageStyle);  
		flexlayout489.addChild(ibUserImage);
		this.ibUserImage = ibUserImage;

		const flexlayout754Style = getCombinedStyle(".flexLayout", {
			width: null,
			height: 60,
			marginLeft: 10,
			id: 6,
			flexGrow: 1
		});  
		var flexlayout754 = new FlexLayout(flexlayout754Style);  
		flexlayout489.addChild(flexlayout754);
		
		const flexlayout861Style = getCombinedStyle(".flexLayout", {
			width: null,
			height: null,
			id: 4,
			flexGrow: 1,
			flexDirection: FlexLayout.FlexDirection.ROW,
			alignItems: FlexLayout.AlignItems.STRETCH
		});  
		var flexlayout861 = new FlexLayout(flexlayout861Style);  
		flexlayout754.addChild(flexlayout861);
		
		const ibUserPositionStyle = getCombinedStyle(".label .label-list-item-title", {
			width: null,
			height: null,
			text: "Manager of S&M",
			textColor: Color.create(128, 74, 74, 74),
			id: 5,
			flexGrow: 1,
			font: Font.create("Arial", 14, Font.NORMAL)
		});  
		var ibUserPosition = new Label(ibUserPositionStyle);  
		if(ibUserPositionStyle.scrollEnabled === false)
			ibUserPosition.ios && (ibUserPosition.ios.scrollEnabled = false);
		flexlayout754.addChild(ibUserPosition);
		this.ibUserPosition = ibUserPosition;

		const ibUserNameStyle = getCombinedStyle(".label .label-list-item-title", {
			width: null,
			height: null,
			text: "Adam Stewart",
			id: 3,
			flexGrow: 2,
			font: Font.create("Arial", 18, Font.NORMAL)
		});  
		var ibUserName = new Label(ibUserNameStyle);  
		if(ibUserNameStyle.scrollEnabled === false)
			ibUserName.ios && (ibUserName.ios.scrollEnabled = false);
		flexlayout861.addChild(ibUserName);
		this.ibUserName = ibUserName;

		const ibUserBirthdayStyle = getCombinedStyle(".label .label-list-item-header", {
			width: null,
			height: null,
			text: "MAY 16",
			textAlignment: TextAlignment.MIDRIGHT,
			id: 2,
			flexGrow: 1,
			font: Font.create("Arial", 14, Font.NORMAL)
		});  
		var ibUserBirthday = new Label(ibUserBirthdayStyle);  
		if(ibUserBirthdayStyle.scrollEnabled === false)
			ibUserBirthday.ios && (ibUserBirthday.ios.scrollEnabled = false);
		flexlayout861.addChild(ibUserBirthday);
		this.ibUserBirthday = ibUserBirthday;

		const imageview159Style = getCombinedStyle(".imageView", {
			width: 30,
			height: 18,
			imageFillType: ImageFillType.ASPECTFIT,
			image: Image.createFromFile("images://icon_birthday.png"),
			id: 1,
			alignSelf: FlexLayout.AlignSelf.CENTER
		});  
		var imageview159 = new ImageView(imageview159Style);  
		flexlayout861.addChild(imageview159);
		
		//assign the children to itemBirthday 
		this.children = Object.assign({}, {
			flexlayout489: flexlayout489,
			ibDivider: ibDivider
		});
		
		//assign the children of flexlayout489
		flexlayout489.children = Object.assign({}, {
			ibUserImage: ibUserImage,
			flexlayout754: flexlayout754
		});
		
		//assign the children of flexlayout754
		flexlayout754.children = Object.assign({}, {
			flexlayout861: flexlayout861,
			ibUserPosition: ibUserPosition
		});
		
		//assign the children of flexlayout861
		flexlayout861.children = Object.assign({}, {
			ibUserName: ibUserName,
			ibUserBirthday: ibUserBirthday,
			imageview159: imageview159
		});
		
	});

ItemBirthday.defaults = getCombinedStyle(".flexLayout", {
	width: 360,
	height: 75,
	paddingLeft: 10,
	paddingRight: 10
});

module && (module.exports = ItemBirthday);
