// library/ItemAnnouncement.js
/* 
		WARNING 
		Auto generated file. 
		Do not modify its contents.
*/

const extend = require('js-base/core/extend');
const FlexLayout = require('sf-core/ui/flexlayout');
const Color = require('sf-core/ui/color');
const Label = require('sf-core/ui/label');
const TextAlignment = require('sf-core/ui/textalignment');
const Font = require('sf-core/ui/font');
const ImageView = require('sf-core/ui/imageview');
const Image = require('sf-core/ui/image');
const ImageFillType = require('sf-core/ui/imagefilltype');



const getCombinedStyle = require("library/styler-builder").getCombinedStyle;

const ItemAnnouncement = extend(FlexLayout)(
	//constructor
	function(_super, props) {
		// initalizes super class for this component scope
		_super(this, props);

		const flexlayout284Style = getCombinedStyle(".flexLayout", {
			width: 80,
			height: null,
			paddingTop: null,
			paddingBottom: null,
			flexGrow: null,
			alignItems: FlexLayout.AlignItems.CENTER,
			justifyContent: FlexLayout.JustifyContent.CENTER
		});
		var flexlayout284 = new FlexLayout(flexlayout284Style);
		this.addChild(flexlayout284);
		
		const flexlayout284_1Style = getCombinedStyle(".flexLayout", {
			width: 10,
			height: null,
			flexGrow: null,
			alignSelf: FlexLayout.AlignSelf.AUTO,
			alignItems: FlexLayout.AlignItems.CENTER,
			justifyContent: FlexLayout.JustifyContent.CENTER
		});
		var flexlayout284_1 = new FlexLayout(flexlayout284_1Style);
		this.addChild(flexlayout284_1);
		
		const flexlayout284_2Style = getCombinedStyle(".flexLayout", {
			width: null,
			height: null,
			marginLeft: 10,
			flexGrow: 1
		});
		var flexlayout284_2 = new FlexLayout(flexlayout284_2Style);
		this.addChild(flexlayout284_2);
		
		const flexlayout446Style = getCombinedStyle(".flexLayout", {
			width: 1,
			height: null,
			left: null,
			top: 0,
			bottom: 0,
			backgroundColor: Color.create(255, 242, 242, 242),
			positionType: FlexLayout.PositionType.ABSOLUTE
		});
		var flexlayout446 = new FlexLayout(flexlayout446Style);
		flexlayout284_1.addChild(flexlayout446);
		
		const label1197Style = getCombinedStyle(".label", {
			width: null,
			height: null,
			text: "11.02.2017",
			textAlignment: TextAlignment.MIDCENTER,
			textColor: Color.create(255, 39, 39, 49),
			marginTop: null,
			flexGrow: 4,
			alignSelf: FlexLayout.AlignSelf.STRETCH,
			font: Font.create("Arial", 14, Font.NORMAL)
		});
		var label1197 = new Label(label1197Style);
		flexlayout284.addChild(label1197);
		
		const label1595Style = getCombinedStyle(".label", {
			width: null,
			height: null,
			text: "Bath County",
			textColor: Color.create(255, 39, 39, 49),
			flexGrow: 4,
			font: Font.create("Arial", 14, Font.NORMAL)
		});
		var label1595 = new Label(label1595Style);
		flexlayout284_2.addChild(label1595);
		
		const label1596Style = getCombinedStyle(".label", {
			width: null,
			height: null,
			text: "Lorem ipsum dolor sit amet, veniam melius duo ei, ea his esse sonet, rebum tation vituperatorib",
			multiline: true,
			textAlignment: TextAlignment.TOPLEFT,
			textColor: Color.create(255, 155, 155, 155),
			flexGrow: 6,
			font: Font.create("Arial", 14, Font.NORMAL)
		});
		var label1596 = new Label(label1596Style);
		flexlayout284_2.addChild(label1596);
		
		const flexlayout1342Style = getCombinedStyle(".flexLayout", {
			width: null,
			height: null,
			flexGrow: 6,
			alignSelf: FlexLayout.AlignSelf.STRETCH
		});
		var flexlayout1342 = new FlexLayout(flexlayout1342Style);
		flexlayout284.addChild(flexlayout1342);
		
		const circleLayoutStyle = getCombinedStyle(".flexLayout", {
			width: null,
			height: null,
			backgroundColor: Color.create(0, 255, 255, 255),
			flexGrow: 4,
			alignSelf: FlexLayout.AlignSelf.STRETCH,
			justifyContent: FlexLayout.JustifyContent.CENTER
		});
		var circleLayout = new FlexLayout(circleLayoutStyle);
		flexlayout284_1.addChild(circleLayout);
		
		const flexlayout446_1Style = getCombinedStyle(".flexLayout", {
			width: null,
			height: null,
			backgroundColor: Color.create(0, 255, 255, 255),
			flexGrow: 6
		});
		var flexlayout446_1 = new FlexLayout(flexlayout446_1Style);
		flexlayout284_1.addChild(flexlayout446_1);
		
		const imageview200Style = getCombinedStyle(".imageView", {
			width: null,
			height: null,
			marginBottom: 5,
			image: Image.createFromFile("images://avatar1.png"),
			imageFillType: ImageFillType.ASPECTFIT,
			flexGrow: 1
		});
		var imageview200 = new ImageView(imageview200Style);
		flexlayout1342.addChild(imageview200);
		
		const circleStyle = getCombinedStyle(".flexLayout", {
			width: 10,
			height: 10,
			borderRadius: 5,
			backgroundColor: Color.create(255, 216, 216, 216),
			alignSelf: FlexLayout.AlignSelf.CENTER
		});
		var circle = new FlexLayout(circleStyle);
		circleLayout.addChild(circle);
		this.circle = circle;

		//assign the children to itemAnnouncement 
		this.children = Object.assign({}, {
			flexlayout284: flexlayout284,
			flexlayout284_1: flexlayout284_1,
			flexlayout284_2: flexlayout284_2
		});
		
		//assign the children of flexlayout284
		flexlayout284.children = Object.assign({}, {
			label1197: label1197,
			flexlayout1342: flexlayout1342
		});
		
		//assign the children of flexlayout284_1
		flexlayout284_1.children = Object.assign({}, {
			flexlayout446: flexlayout446,
			circleLayout: circleLayout,
			flexlayout446_1: flexlayout446_1
		});
		
		//assign the children of flexlayout284_2
		flexlayout284_2.children = Object.assign({}, {
			label1595: label1595,
			label1596: label1596
		});
		
		//assign the children of flexlayout1342
		flexlayout1342.children = Object.assign({}, {
			imageview200: imageview200
		});
		
		//assign the children of circleLayout
		circleLayout.children = Object.assign({}, {
			circle: circle
		});
		
	});

ItemAnnouncement.defaults = getCombinedStyle(".flexLayout", {
	width: 360,
	height: 120,
	flexDirection: FlexLayout.FlexDirection.ROW
});

module && (module.exports = ItemAnnouncement);
