// library/ItemPerformance.js
/* 
		WARNING 
		Auto generated file. 
		Do not modify its contents.
*/

const extend = require('js-base/core/extend');
const FlexLayout = require('sf-core/ui/flexlayout');
const Color = require('sf-core/ui/color');

const ListItem1 = require("../../components/ListItem1");
const HorizontalDivider = require("../../components/HorizontalDivider");

const getCombinedStyle = require("library/styler-builder").getCombinedStyle;

const ItemPerformance = extend(FlexLayout)(
	//constructor
	function(_super, props) {
		// initalizes super class for this component scope
		_super(this, props);

		const reviewDateStyle = getCombinedStyle(".flexLayout", {
			left: 0,
			top: 0,
			width: null,
			height: 45,
			backgroundColor: Color.create(255, 255, 255, 255),
			id: 107,
			flexDirection: FlexLayout.FlexDirection.ROW,
			positionType: FlexLayout.PositionType.RELATIVE,
			alignItems: FlexLayout.AlignItems.CENTER
		});
		var reviewDate = new ListItem1(reviewDateStyle, "itemPerformance");
		this.addChild(reviewDate);
		this.reviewDate = reviewDate;

		const dividerperfStyle = getCombinedStyle(".flexLayout", {
			left: 0,
			top: 0,
			width: null,
			height: 1,
			backgroundColor: Color.create(255, 255, 255, 255),
			alpha: 1,
			id: 106,
			flexGrow: null,
			positionType: FlexLayout.PositionType.RELATIVE
		});
		var dividerperf = new HorizontalDivider(dividerperfStyle, "itemPerformance");
		this.addChild(dividerperf);
		
		const nextReviewStyle = getCombinedStyle(".flexLayout", {
			left: 0,
			top: 0,
			width: null,
			height: 45,
			backgroundColor: Color.create(255, 255, 255, 255),
			id: 105,
			flexDirection: FlexLayout.FlexDirection.ROW,
			positionType: FlexLayout.PositionType.RELATIVE,
			alignItems: FlexLayout.AlignItems.CENTER
		});
		var nextReview = new ListItem1(nextReviewStyle, "itemPerformance");
		this.addChild(nextReview);
		this.nextReview = nextReview;

		const dividerperf_1Style = getCombinedStyle(".flexLayout", {
			left: 0,
			top: 0,
			width: null,
			height: 1,
			backgroundColor: Color.create(255, 255, 255, 255),
			alpha: 1,
			id: 104,
			flexGrow: null,
			positionType: FlexLayout.PositionType.RELATIVE
		});
		var dividerperf_1 = new HorizontalDivider(dividerperf_1Style, "itemPerformance");
		this.addChild(dividerperf_1);
		
		const ratingStyle = getCombinedStyle(".flexLayout", {
			left: 0,
			top: 0,
			width: null,
			height: 45,
			backgroundColor: Color.create(255, 255, 255, 255),
			id: 103,
			flexDirection: FlexLayout.FlexDirection.ROW,
			positionType: FlexLayout.PositionType.RELATIVE,
			alignItems: FlexLayout.AlignItems.CENTER
		});
		var rating = new ListItem1(ratingStyle, "itemPerformance");
		this.addChild(rating);
		this.rating = rating;

		const dividerperf_1_1Style = getCombinedStyle(".flexLayout", {
			left: 0,
			top: 0,
			width: null,
			height: 1,
			backgroundColor: Color.create(255, 255, 255, 255),
			alpha: 1,
			id: 102,
			flexGrow: null,
			positionType: FlexLayout.PositionType.RELATIVE
		});
		var dividerperf_1_1 = new HorizontalDivider(dividerperf_1_1Style, "itemPerformance");
		this.addChild(dividerperf_1_1);
		
		const overallScoreStyle = getCombinedStyle(".flexLayout", {
			left: 0,
			top: 0,
			width: null,
			height: 45,
			backgroundColor: Color.create(255, 255, 255, 255),
			id: 101,
			flexDirection: FlexLayout.FlexDirection.ROW,
			positionType: FlexLayout.PositionType.RELATIVE,
			alignItems: FlexLayout.AlignItems.CENTER
		});
		var overallScore = new ListItem1(overallScoreStyle, "itemPerformance");
		this.addChild(overallScore);
		this.overallScore = overallScore;

		//assign the children to itemPerformance 
		this.children = Object.assign({}, {
			reviewDate: reviewDate,
			dividerperf: dividerperf,
			nextReview: nextReview,
			dividerperf_1: dividerperf_1,
			rating: rating,
			dividerperf_1_1: dividerperf_1_1,
			overallScore: overallScore
		});
		
	});

ItemPerformance.defaults = getCombinedStyle(".flexLayout", {
	width: 309,
	height: 195,
	backgroundColor: Color.create(0, 255, 255, 255),
	paddingBottom: 10
});

module && (module.exports = ItemPerformance);
