// library/LayoutHeaderBar.js
/* 
		WARNING 
		Auto generated file. 
		Do not modify its contents.
*/

const extend = require('js-base/core/extend');
const Label = require('sf-core/ui/label');
const Color = require('sf-core/ui/color');
const TextAlignment = require('sf-core/ui/textalignment');
const Font = require('sf-core/ui/font');
const FlexLayout = require('sf-core/ui/flexlayout');



const getCombinedStyle = require("library/styler-builder").getCombinedStyle;

const LayoutHeaderBar = extend(FlexLayout)(
	//constructor
	function(_super, props) {
		// initalizes super class for this component scope
		_super(this, props);

		const headerBarTitleStyle = getCombinedStyle(".label", {
			backgroundColor: Color.create(0, 255, 255, 255),
			width: null,
			height: 40,
			textAlignment: TextAlignment.MIDCENTER,
			textColor: Color.create(255, 255, 255, 255),
			marginTop: 20,
			flexGrow: null,
			font: Font.create("Arial", 16, Font.NORMAL)
		});
		var headerBarTitle = new Label(headerBarTitleStyle);
		this.addChild(headerBarTitle);
		
		//assign the children to layoutHeaderBar 
		this.children = Object.assign({}, {
			headerBarTitle: headerBarTitle
		});
		
	});

LayoutHeaderBar.defaults = getCombinedStyle(".flexLayout .flexLayout-headerBar", {
	width: 358,
	height: 144,
	positionType: FlexLayout.PositionType.RELATIVE
});

module && (module.exports = LayoutHeaderBar);
