// library/ListItem1.js
/* 
		WARNING 
		Auto generated file. 
		Do not modify its contents.
*/

const extend = require('js-base/core/extend');
const Label = require('sf-core/ui/label');
const Color = require('sf-core/ui/color');
const TextAlignment = require('sf-core/ui/textalignment');
const FlexLayout = require('sf-core/ui/flexlayout');



const getCombinedStyle = require("library/styler-builder").getCombinedStyle;

const ListItem1 = extend(FlexLayout)(
	//constructor
	function(_super, props) {
		// initalizes super class for this component scope
		_super(this, props);

		var title = new Label(getCombinedStyle(".label .label-list-item-title", {
			width: 107,
			height: 50,
			marginLeft: 10,
			text: "Title",
			backgroundColor: Color.create(0, 255, 255, 255)
		})); 
		this.addChild(title);
		
		var value = new Label(getCombinedStyle(".label .label-list-item-value", {
			width: 129,
			height: 50,
			textAlignment: TextAlignment.MIDRIGHT,
			marginRight: 10,
			text: "Value",
			textColor: Color.create(255, 155, 155, 155),
			backgroundColor: Color.create(0, 255, 255, 255),
			flexGrow: 1
		})); 
		this.addChild(value);
		
		//assign the children to listItem1 
		this.children = Object.assign({}, {
			title: title,
			value: value
		});
		
	});

ListItem1.defaults = getCombinedStyle(".flexLayout", {
	left: 12,
	top: 23,
	width: 326,
	height: 59,
	backgroundColor: Color.create(255, 255, 255, 255),
	flexDirection: FlexLayout.FlexDirection.ROW,
	positionType: FlexLayout.PositionType.RELATIVE
});

module && (module.exports = ListItem1);
