// library/styler-builder.js
/* 

!

		WARNING 
		Auto generated file. 
		Do not modify its contents.
		
!

*/
const FlexLayout = require('sf-core/ui/flexlayout');
const Color = require('sf-core/ui/color');
const Image = require('sf-core/ui/image');
const Font = require('sf-core/ui/font');

const styler = require("@smartface/styler/lib/flatStyler");

const ENUMS = {
  "imageFillType": require('sf-core/ui/imagefilltype'),
  "textAlignment": require('sf-core/ui/textalignment'),
  "orientation": require('sf-core/ui/page').Orientation,
  "type": require('sf-core/ui/mapview').Type,
  "gradientOrientation": Color.GradientOrientation,
  "searchViewStyle": require('sf-core/ui/searchview').iOS && require('sf-core/ui/searchview').iOS.style,
  "alignSelf": FlexLayout.AlignSelf,
  "alignContent": FlexLayout.AlignContent,
  "alignItems": FlexLayout.AlignItems,
  "direction": FlexLayout.Direction,
  "flexDirection": FlexLayout.FlexDirection,
  "flexWrap": FlexLayout.FlexWrap,
  "justifyContent": FlexLayout.JustifyContent,
  "positionType": FlexLayout.PositionType,
  "overflow": FlexLayout.OverFlow,
  "style": require('sf-core/ui/statusbarstyle')
};

const COLOR_PROPS = [
  "color",
  "backgroundColor",
  "textColor",
  "borderColor",
  "titleColor",
  "thumbOffColor",
  "thumbOnColor",
  "toggleOffColor",
  "toggleOnColor",
  "hintTextColor",
  "minTrackColor",
  "maxTrackColor",
  "thumbColor"
];

const IMAGE_PROPS = [
  "image",
  "backgroundImage",
  "thumbImage",
  "inactiveImage",
  "maxTrackImage",
  "minTrackImage",
  "backIndicatorImage"
];

const FONT_STYLE = {
  BOLD: "BOLD",
  ITALIC: "ITALIC",
  NORMAL: "NORMAL",
  DEFAULT: "NORMAL"
};

var _styling = {};
var currentTheme = "Defaults";


/**
 * you can register your theme/s 
 * @param themes {Array}
 */
function registerThemes(themes) {
  var _themes = themes;
  if (!(themes instanceof Array))
    _themes = [themes];
  _themes.forEach(function(theme) {
    _styling[theme] = styler(require("../../themes/" + theme + ".json"));
  });
}

/**
 * you can set active theme
 * @param theme
 * @throws Error If theme not registered, throws an error
 */
function setActiveTheme(theme) {
  if (!_styling[theme])
    throw new Error("ensure that theme registered -> " + theme);
  currentTheme = theme;
}

/*
 * get properties of component
 * @param styling 
 * @param className 
 * @return properties object.
 */
function getPropsFromStyle(styling, className) {
  var resultProps = {};

  styling(className)(function(className, key, value) {
    switch (key) {
      case 'flexProps':
        Object.assign(resultProps, getProps(value));
        break;
      default:
        resultProps[key] = getOneProp(key, value);
    }
  });

  return resultProps;

}

/**
 * get combined style for components 
 * @param _className
 * @param _userStyle {optional}
 * @return properties object.
 */
function getCombinedStyle(_className, _userStyle) {
  var styling = _styling[currentTheme];
  var userStyle = Object.assign({}, getPropsFromStyle(styling, _className), _userStyle);
  var resStyle = {};
  Object.keys(userStyle).forEach(function(key) {
    (userStyle[key] !== null) && (resStyle[key] = userStyle[key]);
  });
  return resStyle;
}
//get properties
function getProps(objectVal) {
  var props = {};
  Object
    .keys(objectVal)
    .forEach(function(key) {
      if (objectVal[key] !== null)
        props[key] = getOneProp(key, objectVal[key]);
    });
  return props;
}

//TODO Caching Fonts, Colors
function getOneProp(key, value) {
  var res;
  if (ENUMS[key]) {
    res = ENUMS[key][value];
  }
  else if (COLOR_PROPS.indexOf(key) !== -1) {
    res = createColorForDevice(value);
  }
  else if (IMAGE_PROPS.indexOf(key) !== -1) {
    res = Image.createFromFile("images://" + value);
  }
  else if (key === "font") {
    res = Font.create(value.family || "Arial", value.size || 12, getFontStyle(value));
  }
  else {
    res = value;
  }

  return res;
}

function createColorForDevice(color) {
  var res;
  if (color.startColor) { // gradient color
    res = Color.createGradient({
      startColor: createColorForDevice(color.startColor),
      endColor: createColorForDevice(color.endColor),
      direction: Color.GradientDirection[color.direction]
    });
  }
  else if (/rgb/i.test(color)) {
    var rgba = color.match(/\d\.\d+|\d+/ig);
    res = Color.create((Number(rgba[3]) * 255), Number(rgba[0]), Number(rgba[1]), Number(rgba[2]));
  }
  else {
    res = Color.create(color);
  }
  return res;
}

function getFontStyle(font) {
  var res = "";
  if (font.bold) {
    res += FONT_STYLE.BOLD;
  }
  if (font.italic) {
    res && (res += "_");
    res += FONT_STYLE.ITALIC;
  }!res && (res = FONT_STYLE.DEFAULT);
  return Font[res];
}

module.exports = {
  getCombinedStyle: getCombinedStyle,
  getPropsFromStyle: getPropsFromStyle,
  registerThemes: registerThemes,
  setActiveTheme: setActiveTheme
};
