// library/TopTabItem.js
/* 
		WARNING 
		Auto generated file. 
		Do not modify its contents.
*/

const extend = require('js-base/core/extend');
const Label = require('sf-core/ui/label');
const Color = require('sf-core/ui/color');
const TextAlignment = require('sf-core/ui/textalignment');
const FlexLayout = require('sf-core/ui/flexlayout');



const getCombinedStyle = require("library/styler-builder").getCombinedStyle;

const TopTabItem = extend(FlexLayout)(
	//constructor
	function(_super, props) {
		// initalizes super class for this component scope
		_super(this, props);

		const tabItemTitleStyle = getCombinedStyle(".label", {
			height: null,
			width: null,
			backgroundColor: Color.create(0, 255, 255, 255),
			textAlignment: TextAlignment.MIDCENTER,
			flexGrow: 1
		});
		var tabItemTitle = new Label(tabItemTitleStyle);
		this.addChild(tabItemTitle);
		
		const tabItemBottomLineStyle = getCombinedStyle(".flexLayout", {
			width: null,
			height: 5
		});
		var tabItemBottomLine = new FlexLayout(tabItemBottomLineStyle);
		this.addChild(tabItemBottomLine);
		
		//assign the children to topTabItem 
		this.children = Object.assign({}, {
			tabItemTitle: tabItemTitle,
			tabItemBottomLine: tabItemBottomLine
		});
		
	});

TopTabItem.defaults = getCombinedStyle(".flexLayout", {
	width: 100,
	height: 50,
	backgroundColor: Color.create(0, 255, 255, 255)
});

module && (module.exports = TopTabItem);
